### PR TITLE
allow branch setting on deployment tool repository

### PR DESCRIPTION
**Pull request by seikoudoku2000**
Currently, this script only works under the condition to use `master` of deployment tool repository. But sometimes I want to use a different branch temporally while testing the deployment script. I think this functionality is necessary especially when you have multiple environment hosting goship server, e.g. staging and live.

### CODE REVIEW

File: tools/deploy/deploy.go [hunk 1]
```diff
@@ -11,20 +11,19 @@ import (
...
-	deployProj = flag.String("p", "", "project (required)")
-	deployEnv  = flag.String("e", "", "environment (required)")
-	configFile = flag.String("g", "/tmp/deploy.yaml", "shared config setting ( default /tmp/deploy.yaml)")
-	pullOnly   = flag.Bool("o", false, "chef update only (default false)")
-	skipUpdate = flag.Bool("m", false, "skip the chef update (default false)")
-	bootstrap  = flag.Bool("b", false, "bootstrap a server ( default false)")
+	deployProj       = flag.String("p", "", "project (required)")
+	deployEnv        = flag.String("e", "", "environment (required)")
+	configFile       = flag.String("g", "/tmp/deploy.yaml", "shared config setting ( default /tmp/deploy.yaml)")
+	deployToolBranch = flag.String("d", "master", "deploy tool branch ( default master)")
```
Comment (seikoudoku2000) [inline tools/deploy/deploy.go]: Added this parameter.

File: tools/deploy/deploy.go [hunk 2]
```diff
@@ -71,35 +70,23 @@ func parseConfig() (c config) {
...
+	log.Println("Updating devops-tools")
+	os.Setenv("GIT_SSH", "/tmp/private_code/wrap-ssh4git.sh")
+	gitCheckoutCmd := "/usr/bin/git --git-dir=" + conf.chefRepo + "/.git --work-tree=" + conf.chefRepo + " checkout " + *deployToolBranch
+	// TODO: refactor "execCmd" and run commands at once
+	_, err := execCmd(gitCheckoutCmd, conf)
 	if err != nil {
-		log.Fatal("ERROR:  failed to get commits from GitHub: Please try again later ", err)
+		log.Fatal("ERROR: Failed to checkout: ", err)
 	}
-	remoteHash := *commits[0].SHA
```
Comment (seikoudoku2000) [inline tools/deploy/deploy.go]: Removed comparison with remote branch because always pulling the latest would work.

**Comment by hayate** [comment]
makes sense LGTM

**Comment by toshipon** [comment]
LGTm

**Pull request merged**